doubtfire-lms/doubtfire-web
# Migrate/confirmation-modal component

Original confirmation-modal migration PR: https://github.com/doubtfire-lms/doubtfire-web/pull/846

Created a new branch from the development branch and added @s223749059's commits before making additional changes of replacing the bootstrap html with material components and a couple fixes.

# Description

Migrates the `confirmation-modal` component from CoffeeScript to TypeScript. This component is used in the unit-tutorials-list, unit-task-editor, and portfolio-review-step components. The confirmation modal is used to display a dialog with a title, message, where users can either click cancel to dismiss the dialog, or click confirm which will run the callback function originally passed when opening the modal.

## Type of change

 
# How Has This Been Tested?

- Login as `aconvenor` or `aadmin`
- In the top right, click Mange Units
- Select a unit
- Select the Tasks tab
- Click on the delete icon to be prompted with the confirmation modal
- Click cancel to dismiss the modal, or click Confirm to delete the unit task

## Before:
![confrimation-modal-before](https://github.com/user-attachments/assets/c9a7f80f-1c97-4a59-8ed9-5935b24d630f)

## After:
![confirmation-modal-after](https://github.com/user-attachments/assets/080eea38-d17e-4d21-a4b7-e1e18a539af9)


## Testing Checklist:


# Checklist:
